src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -1842,8 +1816,7 @@ void Assembler::cmovl(Condition cc, Register dst, Register src) {
 
 void Assembler::ecmovl(Condition cc, Register dst, Register src1, Register src2) {
   InstructionAttr attributes(AVX_128bit, /* vex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
-  int encode = evex_prefix_and_encode_ndd(src1->encoding(), dst->encoding(), src2->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes);
-  emit_int16((0x40 | cc), (0xC0 | encode));
+  evex_opcode_and_int16_ndd(dst->encoding(), src1->encoding(), src2->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, 0x40 | cc, 0xC0, false);
```

Comment:
This could be instead call to:
evex_opcode_int16_ndd(dst->encoding(), src2->encoding(), src1->encoding(), ..)
We can remove the separate method evex_opcode_and_int16_ndd() method altogether.

Reply:
Actually, I tried implementing using only one function but due to the argument ordering changes in the inner functions, it was not possible to do it using one high level function like `evex_opcode_int16_ndd`.